scripts/spec/uaa_sync_admin_users_spec.rb

```diff
@@ -0,0 +1,303 @@
+            to_return(
+              status: 200,
+              headers: {"content-type" => "application/json"},
+              body: json_group_response(group_name, ["admin", "existing_user", "new_user", "removed_user"])
+          )
+        }
+
+        WebMock.stub_request(:delete, "#{target}/Users/__user__removed_user__uuid__").
+          to_return(
+            status: 200,
+            headers: {"content-type" => "application/json"},
+            body: json_user_response("removed_user", "removed_user@example.com"),
+        )
+
+        @created_users, @deleted_users = uaa_sync_admin_users.update_admin_users(users)
+
+      end
+
+      it "creates only the new users" do
+        expect(WebMock).to have_requested(
+          :post, "#{target}/Users"
+        ).with(
+          body: /"userName":"new_user"/,
+        ).once
+        expect(WebMock).not_to have_requested(
+          :post, "#{target}/Users"
+        ).with(
+          body: /"userName":"existing_user"/,
+        ).once
+      end
+
+      it "new users are added as members of admin groups" do
+        uaa_sync_admin_users.admin_groups.each { |groupname|
+          expect(WebMock).to have_requested(
+            :put, "#{target}/Groups/__group__#{groupname}__uuid__"
+          ).with(
+            body: /"members":.*"__user__new_user__uuid__"/,
+          ).once
+        }
+      end
+
+      it "existing members and admin are kept in the groups" do
+        uaa_sync_admin_users.admin_groups.each { |groupname|
+          expect(WebMock).to have_requested(
+            :put, "#{target}/Groups/__group__#{groupname}__uuid__"
+          ).with(
+            body: /"members":.*"__user__existing_user__uuid__"/,
+          ).once
+          expect(WebMock).to have_requested(
+            :put, "#{target}/Groups/__group__#{groupname}__uuid__"
+          ).with(
+            body: /"members":.*"__user__admin__uuid__"/,
+          ).once
+        }
+      end
+      it "does not delete the admin user" do
+        expect(WebMock).not_to have_requested(
+          :delete, "#{target}/Users/__user__admin__uuid__"
+        )
+      end
+
+      it "deletes extra users older than 2h" do
+        expect(WebMock).to have_requested(
+          :delete, "#{target}/Users/__user__removed_user__uuid__"
+        )
+      end
+
+      it "keeps extra users newer than 2h" do
+        expect(WebMock).not_to have_requested(
+          :delete, "#{target}/Users/__user__test_user_1__uuid__"
+        )
+        expect(WebMock).not_to have_requested(
+          :delete, "#{target}/Users/__user__test_user_2__uuid__"
+        )
+      end
+
+      it "returns created and deleted users" do
+        expect(@created_users.length).to be(1)
+        expect(@created_users[0]).to include(:username => "new_user")
+        expect(@deleted_users.length).to be(1)
+        expect(@deleted_users[0]).to include(:username => "removed_user")
+      end
+    end
+  end
+
+end
+
+def user_uuid(username)
```

Comment:
It would be good for these helper methods to be scoped inside the `describe`. This will prevent them from polluting the global namespace and potentially clashing with other tests.